lib/block-patterns.php

```diff
@@ -236,15 +236,52 @@ function load_remote_patterns() {
 	}
 }
 
+/**
+ * Register `Featured` (category) patterns from wordpress.org/patterns.
+ *
+ * @since 5.9.0
+ */
+function gutenberg_load_remote_featured_patterns() {
+	/**
+	 * Filter to disable remote block patterns.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param bool $should_load_remote
+	 */
+	$should_load_remote = apply_filters( 'should_load_remote_block_patterns', true );
+
+	if ( ! $should_load_remote ) {
+		return;
+	}
+
+	if ( ! WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'featured' ) ) {
+		register_block_pattern_category( 'featured', array( 'label' => __( 'Featured', 'gutenberg' ) ) );
+	}
+	$request             = new WP_REST_Request( 'GET', '/wp/v2/pattern-directory/patterns' );
+	$request['category'] = 26; // This is the `Featured` category id from pattern directory.
+	$response            = rest_do_request( $request );
+	if ( $response->is_error() ) {
+		return;
+	}
+	$patterns = $response->get_data();
+	foreach ( $patterns as $pattern ) {
+		$pattern_name = sanitize_title( $pattern['title'] );
+		if ( ! WP_Block_Patterns_Registry::get_instance()->is_registered( $pattern_name ) ) {
+			register_block_pattern( $pattern_name, (array) $pattern );
+		};
```

Comment:
Some of these same patterns are already being registered with the prefix `core/` and there's no prefix here, so I'm seeing a few duplicates in the featured tab: `core/heading` and `heading`, `core/large-header-with-left-aligned-text` and `large-header-with-left-aligned-text`, etc.

Reply:
hm... I updated with a check if a pattern is registered with the `core` prefix. This made me wonder why we need the `core` prefix in the first place.. Should we also incorporate the `id` that exists in pattern directory in [is_registered](https://developer.wordpress.org/reference/classes/wp_block_patterns_registry/is_registered/)?